### PR TITLE
Parser

### PR DESCRIPTION
**Pull request by radavid2017**
Features:
- [x] Error Handling
- [x] Google Tests

### CODE REVIEW

File: src/packageParser.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        // -----------   Code details   -----------
+        // '\\d+' matches one or more decimal digits (0-9)
+        // '\\.' matches a literal dot. Backslash is used to escape the dot because the dot has a special meaning in regular expressions (matches any character)
+        // "(\\d+\\.\\d+)" all together matches decimal numbers with one or more digits before and after the dot. ex: "12.34"
+
+        std::regex regex("(\\d+\\.\\d+)");
+        std::sregex_iterator iter(data.begin(), data.end(), regex);
+        std::sregex_iterator end;
+        int idx = 0;
+        while (iter != end && idx < 4) {
```
Comment (marisvali) [inline src/packageParser.cpp:102]: Same as above, the while is not necessary.

**Review by marisvali — COMMENTED** [review]
Found a few things but I have to stop the review now and switch to another task.

File: src/packageParser.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        // -----------   Code details   -----------
+        //    '\\d+' matches one or more decimal digits (0-9)
+        //    '\\.' matches a literal dot. Backslash is used to escape the dot because the dot has a special meaning in regular expressions (matches any character)
+        //    "(\\d+\\.\\d+)" all together matches decimal numbers with one or more digits before and after the dot. ex: "12.34"
+
+        std::regex regex("(\\d+\\.\\d+)");
+        std::sregex_iterator iter(data.begin(), data.end(), regex);
+        std::sregex_iterator end;
+        int idx = 0;
+        while (iter != end && idx < 8) {
```
[the remark this answers is not in the document]
Comment (radavid2017) [reply src/packageParser.cpp]: Done

File: src/packageParser.cpp (hunk 1 above)
Comment (radavid2017) [reply src/packageParser.cpp]: Done

File: include/packageParser.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    // PackageParser HEADER
+
+    class PackageParser {
+    public:
+        // Parses the entire 'top' command output and populates the TopPackage object
+        data::TopPackage ParseData(const std::string& data);
+
+        // Parses the CPU usage information and returns the CpuUsage object
+        data::CpuUsage ParseCpuUsage(const std::string& data);
```
[the remark this answers is not in the document]
Comment (radavid2017) [reply include/packageParser.h:18]: Done

**Review by CristianIonutIonescu — APPROVED** [review]

**Pull request merged**